### PR TITLE
Bps0027 mvc entity framework

### PR DESCRIPTION
**Pull request by fbensibpsolutions**
## General Description
asana issue : https://app.asana.com/0/1111428436201563/1200977317635938/f

#### Add New Customer
1. Click "Customer" button at the top of the header.
2. Fill-up all input fields.
3. Don't leave empty fields when we try to save new customer.
4. Click "Save" button if all fields are not empty fields.

#### Edit Customer
1. Click "Edit" link button to redirect you from edit customer.
2. All input fields are required not to leave empty fields.
3. Click "Edit" button to continue update new data.

### Delete Customer
1. Click "Delete" link button to redirect you from delete customer.
2. Check first the customer name before we continue press the delete button, It will no longer exists in customer lists after click the button delete.

### CODE REVIEW

File: BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230026141_CustomersTable.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        public override void Up()
+        {
+        }
+        
+        public override void Down()
+        {
+        }
+    }
+}
```
Comment (jrvalenzona-bps) [inline BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230026141_CustomersTable.cs]: Remove unused migration

File: BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230014448_InitialCreate.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    {
+                        CustomerId = c.Int(nullable: false, identity: true),
+                        No = c.String(),
+                        Date = c.String(),
+                        CustomerName = c.String(),
+                        Amount = c.Int(nullable: false),
+                    })
+                .PrimaryKey(t => t.CustomerId);
+            
+        }
```
Comment (jrvalenzona-bps) [inline BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230014448_InitialCreate.cs]: Please use Fluent API In Code First Approach.

**Comment by jrvalenzona-bps** [comment]
To make it clean on database side, please delete your database on server and remove also your migration. you need to create a new database using a Fluent API In Code First Approach.

File: BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230026141_CustomersTable.cs (hunk 1 above)
Comment (fbensibpsolutions) [reply BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230026141_CustomersTable.cs]: Done.

File: BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230014448_InitialCreate.cs (hunk 2 above)
Comment (fbensibpsolutions) [reply BPS0026_MVC_WEB/BPS0026_MVC_WEB/Migrations/202109230014448_InitialCreate.cs]: Done.
Salamat sa advice :)

File: BPS0026_MVC_WEB/BPS0026_MVC_WEB/Context/CustomerContext.cs [hunk 3]
```diff
@@ -10,5 +11,23 @@ namespace BPS0026_MVC_WEB.Context
...
+            modelBuilder.Entity<Customers>().Property(c => c.No).IsMaxLength();
+
+            modelBuilder.Entity<Customers>().Property(c => c.Date).HasMaxLength(10);
+
+            modelBuilder.Entity<Customers>().Property(c => c.CustomerName).HasMaxLength(50);
+
+            modelBuilder.Entity<Customers>().Property(c => c.Amount);
+
+            base.OnModelCreating(modelBuilder);
+        }
```
Comment (jrvalenzona-bps) [inline BPS0026_MVC_WEB/BPS0026_MVC_WEB/Context/CustomerContext.cs]: Add modelBuilder.Entity<Customers>().ToTable("Customers").
Change Customers model to Customer (singular).
Change Date property to DateTime2

Comment (fbensibpsolutions) [reply BPS0026_MVC_WEB/BPS0026_MVC_WEB/Context/CustomerContext.cs]: Done .

**Pull request merged**